marketplacetests/tests/test_marketplace_forgot_pin.py

```diff
@@ -14,23 +14,19 @@ class TestMarketplaceForgotPin(MarketplaceGaiaTestCase):
 
     def test_forgot_pin(self):
 
-        app_name = 'Test Zippy With Me'
         old_pin = '1234'
         new_pin = '1111'
-        acct = FxATestAccount(base_url=self.base_url).create_account()
-
-        if self.apps.is_app_installed(app_name):
-            raise Exception('The app %s is already installed.' % app_name)
 
         marketplace = Marketplace(self.marionette, self.MARKETPLACE_DEV_NAME)
         home_page = marketplace.launch()
 
-        home_page.login(acct.email, acct.password)
+        app_name = home_page.find_first_paid_app()
```

Comment:
As above, searching just once would be much more efficient, unless there's a reason for multiple searches? If there is, perhaps the first could be via the API or headless request? (I'm fine with this being addressed separately).